shared/services/rocketpool/client.go

```diff
@@ -1635,6 +1636,63 @@ func (c *Client) composeAddons(cfg *config.RocketPoolConfig, rocketpoolDir strin
 		deployedContainers = append(deployedContainers, filepath.Join(overrideFolder, graffiti_wall_writer.GraffitiWallWriterContainerName+composeFileSuffix))
 	}
 
+	// APCUPSD
+	if cfg.EnableMetrics.Value == true && cfg.Apcupsd.GetEnabledParameter().Value == true {
+		runtimeFolder := filepath.Join(rocketpoolDir, runtimeDir, "addons", "apcupsd")
+		templatesFolder := filepath.Join(rocketpoolDir, templatesDir, "addons", "apcupsd")
+		overrideFolder := filepath.Join(rocketpoolDir, overrideDir, "addons", "apcupsd")
+
+		// Make the addon folder
+		err := os.MkdirAll(runtimeFolder, 0775)
+		if err != nil {
+			return []string{}, fmt.Errorf("error creating addon runtime folder (%s): %w", runtimeFolder, err)
+		}
+
+		// Write container file
+		apcupsdMode := cfg.Apcupsd.GetConfig().GetParameters()[1]
+
+		var containerContents []byte
+
+		if apcupsdMode.Value == apcupsd.Mode_Network {
```

Comment:
This is perhaps a bit novel as I conditionally choose which compose file to use based on the users preference on how to run APCUPSD. I've tested and the smartnode seems to create and destroy the services correctly when toggling between the two different setups.